scripts/export_requirements.py

```diff
@@ -55,7 +57,86 @@ def get_hash(content: dict) -> str:
     return hash == get_hash(content)
 
 
-def main(req_path: os.PathLike, should_validate_hash: bool = True) -> typing.Optional[int]:
+def _extract_packages_from_requirements(requirements: str) -> list:
+    """Extract a list of packages from the provided requirements str."""
+    req = requirements.split("\n")
+    packages = set()
+    for line in req.copy():
+        if line.startswith("#"):
+            continue
+        if not len(line.strip()):
+            continue
+
+        # requirement files we will be parsing can have `;` or =<>~
+        match = PACKAGE_REGEX.match(line)
+        if match is None:
+            continue
+
+        package = match.group("package").strip().lower().replace("_", "-")
+        packages.add(package)
+    return packages
+
+
+def _update_versions_in_requirements(requirements: str, packages: dict) -> str:
+    """Update the versions in requirements with the provided package to version mapping."""
+    req = requirements.split("\n")
+    for i, line in enumerate(req.copy()):
+        if line.startswith("#"):
+            continue
+        if not len(line.strip()):
+            continue
+
+        # requirement files we will be parsing can have `;` or =<>~
+        match = PACKAGE_REGEX.match(line)
+
+        package = match.group("package").strip()
+        try:
+            req[i] = package + "==" + packages[package.lower().replace("_", "-")]
```

Comment:
Wouldn't the package already be lower cased and have no `_` as we do these changes on line 76.

Reply:
These are different functions.

Reply:
Ah yes, I thought packages is coming from that function...which isn't the case.

But when we form this `packages` variable, can we just take the key as the edited name (on line 125), this would also reduce the redundancy to run the edit twice.

Reply:
Made the above function return a result which is used on this function.